src/vs/code/browser/issue/issue.ts

```diff
@@ -0,0 +1,1205 @@
+		const issueType = this.issueReporterModel.getData().issueType;
+
+		if (this.loadingExtensionData) {
+			return false;
+		}
+
+		if (this.isWeb) {
+			if (issueType === IssueType.FeatureRequest || issueType === IssueType.PerformanceIssue || issueType === IssueType.Bug) {
+				return true;
+			}
+		} else {
+			if (issueType === IssueType.Bug && this.receivedSystemInfo) {
+				return true;
+			}
+
+			if (issueType === IssueType.PerformanceIssue && this.receivedSystemInfo && this.receivedPerformanceInfo) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private getExtensionRepositoryUrl(): string | undefined {
+		const selectedExtension = this.issueReporterModel.getData().selectedExtension;
+		return selectedExtension && selectedExtension.repositoryUrl;
+	}
+
+	public getExtensionBugsUrl(): string | undefined {
+		const selectedExtension = this.issueReporterModel.getData().selectedExtension;
+		return selectedExtension && selectedExtension.bugsUrl;
+	}
+
+	public searchVSCodeIssues(title: string, issueDescription?: string): void {
+		if (title) {
+			this.searchDuplicates(title, issueDescription);
+		} else {
+			this.clearSearchResults();
+		}
+	}
+
+	public searchIssues(title: string, fileOnExtension: boolean | undefined, fileOnMarketplace: boolean | undefined): void {
+		if (fileOnExtension) {
+			return this.searchExtensionIssues(title);
+		}
+
+		if (fileOnMarketplace) {
+			return this.searchMarketplaceIssues(title);
+		}
+
+		const description = this.issueReporterModel.getData().issueDescription;
+		this.searchVSCodeIssues(title, description);
+	}
+
+	private searchExtensionIssues(title: string): void {
+		const url = this.getExtensionGitHubUrl();
+		if (title) {
+			const matches = /^https?:\/\/github\.com\/(.*)/.exec(url);
+			if (matches && matches.length) {
+				const repo = matches[1];
+				return this.searchGitHub(repo, title);
+			}
+
+			// If the extension has no repository, display empty search results
+			if (this.issueReporterModel.getData().selectedExtension) {
+				this.clearSearchResults();
+				return this.displaySearchResults([]);
+
+			}
+		}
+
+		this.clearSearchResults();
+	}
+
+	private searchMarketplaceIssues(title: string): void {
+		if (title) {
+			const gitHubInfo = this.parseGitHubUrl(this.product.reportMarketplaceIssueUrl!);
+			if (gitHubInfo) {
+				return this.searchGitHub(`${gitHubInfo.owner}/${gitHubInfo.repositoryName}`, title);
+			}
+		}
+	}
+
+	public async close(): Promise<void> {
+		await this.issueMainService.$closeReporter();
+	}
+
+	public clearSearchResults(): void {
+		const similarIssues = this.getElementById('similar-issues')!;
+		similarIssues.innerText = '';
+		this.numberOfSearchResultsDisplayed = 0;
+	}
+
+	@debounce(300)
+	private searchGitHub(repo: string, title: string): void {
+		const query = `is:issue+repo:${repo}+${title}`;
+		const similarIssues = this.getElementById('similar-issues')!;
+
+		fetch(`https://api.github.com/search/issues?q=${query}`).then((response) => {
```

Comment:
hmm that's a good point. I think the github API in this situation is okay since the "query" will just search for `potentially random repo` + `potentially random title`, which would come up with an `OK` response since it's a valid API call with that link and format, but just wouldn't have search results